Vim/Nvim plugin related fixes

enggnr opened the pull request:
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This fixes issues identified during the setup of Vim/Nvim plugins.

- **Other information**:
Fixes #123 and #127. Related to #https://github.com/megabyte-labs/AstronVim/pull/1

ProfessorManhattan commented:
Hey @enggnr --- I manually merged the changes so far... I notice that when I run:

```
bash -ci "nvim --headless -c 'quitall'"
```

I see:

```
mason.nvim has not been set up. Make sure to set up 'mason' before 'mason-lspconfig'. :h mason-lspconfig-quickstart`
```

Also, after the command finishes, when I open `nvim` it shows a bunch of unstyled notifications that say `mason-null-ls` and a few names. I wonder, is `mason-null-ls` an issue? There's something null? Anyway, I'd like to finish the entire boot process of `nvim` headlessly. It looks like `vim` is configured properly now but `nvim` needs some work to headlessly install all the mason languages. Also, can we do anything to improve the start-out experience for nvim? I think for starters, we should have the file tree window open by default, with the initial cursor focus on the text file that's open.

Also, is there a better way of running bash -ci "nvim --headless -c 'quitall'"? i.e. can we do it without an interactive Bash session?

enggnr commented:
> mason.nvim has not been set up. Make sure to set up 'mason' before 'mason-lspconfig'. :h mason-lspconfig-quickstart`

It was caused by the presence of `.config/npm/config/npm-init.js` file. Removing this fixed the problems. I am trying to find what this files is used for.

Note that even though the console says `Mason is installing packages if configured, check status with :Mason`, after `nvim` is opened and you use Mason any time, it installs a bunch of items. There is no background way to do it as far as I could tell. Mason does things in the background and this could be why there is no _silent_ install.

enggnr commented:
@ProfessorManhattan, this has been addressed. Problem was caused by the file `dot_config/npm/config/npm-init.js`. I was not able to find what this file does, I've removed that file and `dot_config/npm/favourites.yml`. If these are needed, we would need to address the failures caused due to these files (mostly the js file).

To show file tree, I've updated the setting in https://github.com/megabyte-labs/AstronVim/commit/2f60dbeaddc9efddeb6202295824dc0b46d7b709.

ProfessorManhattan commented:
Ok good stuff. Thanks

On Thu, Feb 15, 2024, 7:39 AM enggnr ***@***.***> wrote:

> @ProfessorManhattan <https://github.com/ProfessorManhattan>, this has
> been addressed. Problem was caused by the file
> dot_config/npm/config/npm-init.js. I was not able to find what this file
> does, I've removed that file and dot_config/npm/favourites.yml. If these
> are needed, we would need to address the failures caused due to these files
> (mostly the js file).
>
> To show file tree, I've updated the setting in megabyte-labs/AstronVim@
> 2f60dbe
> <https://github.com/megabyte-labs/AstronVim/commit/2f60dbeaddc9efddeb6202295824dc0b46d7b709>
> .
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/megabyte-labs/install.doctor/pull/135#issuecomment-1946019176>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOJRHXIJIHTMBWDGJAQVBQ3YTX6WRAVCNFSM6AAAAABBEPSHACVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNBWGAYTSMJXGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
